Tests/MapboxCoreNavigationTests/NavigationServiceTests.swift

```diff
@@ -755,9 +758,56 @@ class NavigationServiceTests: TestCase {
         wait(for: [rerouteTriggeredExpectation, didRerouteExpectation], timeout: locationManager.expectedReplayTime)
         locationManager.stopUpdatingLocation()
         
+        XCTAssertFalse(delegate.recentMessages.contains("navigationService(_:willModify:)"))
         XCTAssertTrue(delegate.recentMessages.contains("navigationService(_:didRerouteAlong:at:proactive:)"))
     }
 
+    func testReroutingUpdatesRouteOptions() {
+        let trace = Fixture.generateTrace(for: initialRouteResponse.routes!.first!).shiftedToPresent()
+        guard let firstLoction = trace.first,
+              let lastLocation = trace.last,
+              firstLoction != lastLocation else {
+                  XCTFail("Invalid trace"); return
+              }
+        let locationManager = ReplayLocationManager(locations: trace)
+        locationManager.speedMultiplier = 100
+        let service = MapboxNavigationService(routeResponse: initialRouteResponse,
+                                              routeIndex: 0,
+                                              routeOptions: routeOptions,
+                                              customRoutingProvider: nil,
+                                              credentials: Fixture.credentials,
+                                              locationSource: locationManager)
+        service.delegate = delegate
+        let router = service.router
+        
+        let finishExpectation = expectation(description: "Reroute should finish either way.")
```

Comment:
This is not an ideal solution, because default implementation will do an online request with undetermined timeout. For the test we need some notification to tell that request has actually happened, but we don't care about it's result.
I'm afraid this could be another flaky test, but don't have good ideas about how to hook up the check...

Reply:
seems to be worked around by different waiting approach